api/tests/unit/application/users/user-controller_test.js

```diff
@@ -90,9 +93,28 @@ describe('Unit | Controller | user-controller', () => {
   });
 
   describe('#getAuthenticatedUserProfile', () => {
-    it('should be a function', () => {
+    const expectedSerializedProfile = Symbol('a serialized profile');
+    const userId = 1234;
+
+    beforeEach(() => {
+      sinon.stub(tokenService, 'extractUserId').returns(userId);
+
+      const aUser = new User({ id: userId });
```

Comment:
Pourquoi utiliser un vrai `User` du domain et pas juste un simple `const aUser = { userId }` ? Et pourquoi ne pas utiliser le `domainBuilder` (qui peut sembler overkill mais bon, si on décide d'utiliser des vrais objets du domain dans les TU, tant qu'à faire)

Reply:
J'utilise la classe `User` pour fournir le bon type. Comme on a des objets métier faciles à instancier pour la plupart (ce qui est une bonne chose), ça me paraît bien.

Effectivement, je ne l'ai pas fait avec les `Symbols` donc c'est pas très cohérent, intéressant.

Je note aussi que j'ai pas eu besoin du `domainBuilder`, intéressant aussi. C'est soit que le `User` n'a pas d'exigences autres que "avoir un id", soit que le constructeur ne les vérifie pas.

Je pourrais passer par le `domainBuilder` par souci d'homogénéité, intéressant aussi, merci.

Reply:
En fait j'ai du mal à comprendre ce que tu entends par passer fournir le bon type. Plus spécifiquement, j'aimerai bien voir un exemple de test unitaire où on aurait besoin de passer un vrai `User` (à savoir si on ne le faisait pas, il y aurait un vrai problème). 

Les deux seuls cas que j'ai en tête où c'est effectivement nécessaire sont du code où on aurait : 
- des `if (user instanceof User) { ... }`
- des méthodes de l'instances qui seraient appelées et qu'on ne souhaiterait pas stubber 

Dans les deux cas, ça me semble soit un code smell (`instanceof`), soit un test plus vraiment unitaire. Et aucun de ces deux cas n'est présent sur la codebase (il me semble). 

Intuitivement je me dis que c'est bien aussi de passer les objets du bon type mais concrètement je trouve aucun argument qui le justifie vraiment. C'est pour ça que j'avais proposer de supprimer le dossier `domain-builder` mais ça n'avait enthousiasmé personne autre que moi :D

Reply:
Pour moi c'est deux idées :
- fournir le type réel,
- pour un type donné, par exemple `User`, il y a un certain nombre de choses qui doivent être toujours vraies (invariants).

Derrière fournir le type réel, j'ai une TypeScript envy, donc ça va dans cette direction.

Pour les invariants (métier), on n'en a pas beaucoup actuellement, mais par exemple si on regarde `User`, un user doit toujours être capable de dire s'il a le rôle PixMaster.

Passer par le constructeur de `User` quand je peux me permet de rendre le test indépendant de l'usage ou non de ces invariants dans l'implémentation de ce que je teste.